dask/dataframe/io.py

```diff
@@ -724,10 +736,57 @@ def one_path_one_key(path, key, start, stop, columns, chunksize, lock):
     if (start != 0 or stop is not None) and len(keys) > 1:
         raise NotImplementedError(read_hdf_error_msg)
     from .multi import concat
-    return concat([one_path_one_key(path, k, start, s, columns, chunksize, lock)
+    return concat([one_path_one_key(path, k, start, s, columns, chunksize,
+                                    lock, division_column)
                    for k, s in zip(keys, stops)])
 
 
+def _find_divisions(division_column, start, stop, chunksize, path, key):
+    divisions = list()
+    starts = [start]
+    stops = list()
+    row = start
+    stop = pd.HDFStore(path).get_storer(key).nrows if stop is None else stop
+    data = _read_hdf_column(path, key, division_column, row, 1)
+    divisions.append(data[0])
+    for row in range(start + chunksize, stop, chunksize):
+        data = _read_hdf_column(path, key, division_column, row - 1, 100)
+        split = _find_split(data)
+        while not split and row < stop:
+            row = row + 100
+            data = _read_hdf_column(path, key, division_column, row - 1, 100)
+            if len(data) == 0:
+                split = False
+            else:
+                split = _find_split(data)
+        if split > 0:
+            starts.append(row)
+            stops.append(row)
+            divisions.append(data[split])
+    stops.append(stop)
+    data = _read_hdf_column(path, key, division_column, stop - 1, 1)
+    divisions.append(data[0])
+    return divisions, starts, stops
+
+
+def _find_split(data):
+    i = 0
+    while data[i] == data[i + 1]:
+        i += 1
+        if i == len(data) - 1:
+            return False
+    return i + 1
+
+
+def _read_hdf_column(path, key, column, start, rows):
+    if isinstance(column, string_types):
+        return pd.HDFStore(path).select(key, start=start, stop=start + rows,
+                                        columns=[column])[column].values
```

Comment:
if you are selecting a single columns, use ``.select_column``, you cannot use a where clauses, but ``start/stop`` will work. This will be MUCH faster as the ``columns`` kw selects ALL columns then does a reindex (its how PyTables works).